test/tools/unified-spec-runner/match.ts

```diff
@@ -321,9 +330,12 @@ export function expectErrorCheck(
   expected: ExpectedError,
   entities: EntitiesMap
 ): boolean {
-  if (Object.keys(expected)[0] === 'isClientError' || Object.keys(expected)[0] === 'isError') {
-    // FIXME: We cannot tell if Error arose from driver and not from server
-    return;
+  if (expected.isClientError === true) {
+    expect(error).to.satisfy(
+      (error: unknown) => error instanceof MongoDriverError || error instanceof MongoNetworkError
+    );
+  } else if (expected.isClientError === false) {
+    expect(error).to.be.instanceOf(MongoServerError);
```

Comment:
This was possible to fix since we merged the last of the better errors project. But now we can differentiate clearly `isClientError`s and server errors. Even though MongoNetworkError aren't descendent of MongoDriverError we consider it to be a client error here since we are the one to capture and construct it, as opposed to MongoServerError being made of message and code directly from MongoDB.

Reply:
Not sure I follow, `isClientError` is also optional. Are you saying that in situations where only `isError` is specified, and `isClientError` is not specified, we can't assert one of the above specific types?

Reply:
Yes, I think the test doesn't care about the origin of the error if `isClientError` is omitted, so we shouldn't assert anything.